biodatasets/n2c2_2018_track1/n2c2_2018_track1.py

```diff
@@ -0,0 +1,282 @@
+            version=SOURCE_VERSION,
+            description=_DATASETNAME + " source schema",
+            schema=SOURCE,
+            subset_id=_DATASETNAME,
+        ),
+        BigBioConfig(
+            name=_BIGBIO_CONFIG_NAME,
+            version=BIGBIO_VERSION,
+            description=_DATASETNAME + " BigBio schema",
+            schema=BIGBIO_TEXT,
+            subset_id=_DATASETNAME,
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = _SOURCE_CONFIG_NAME
+    LABEL_CLASS_NAMES = [
+        "ABDOMINAL",
+        "ADVANCED-CAD",
+        "ALCOHOL-ABUSE",
+        "ASP-FOR-MI",
+        "CREATININE",
+        "DIETSUPP-2MOS",
+        "DRUG-ABUSE",
+        "ENGLISH",
+        "HBA1C",
+        "KETO-1YR",
+        "MAJOR-DIABETES",
+        "MAKES-DECISIONS",
+        "MI-6MOS",
+    ]
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == SOURCE:
+            labels = {
+                key: datasets.ClassLabel(names=["met", "not met"])
+                for key in N2C22018CohortSelectionDataset.LABEL_CLASS_NAMES
+            }
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "document_id": datasets.Value("string"),
+                    "text": datasets.Value("string"),
+                    "tags": labels,
+                }
+            )
+
+        elif self.config.schema == BIGBIO_TEXT:
+            features = schemas.text.features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager) -> List[datasets.SplitGenerator]:
+
+        if self.config.data_dir is None or self.config.name is None:
+            raise ValueError("This is a local dataset. Please pass the data_dir and name kwarg to load_dataset.")
+        else:
+            data_dir = self.config.data_dir
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "file_path": os.path.join(data_dir, "train.zip"),
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={
+                    "file_path": os.path.join(data_dir, "n2c2-t1_gold_standard_test_data.zip"),
+                },
+            ),
+        ]
+
+    @staticmethod
+    def _get_source_sample(sample_id, sample):
+        return {
+            "id": sample_id,
+            "document_id": sample_id,
+            "text": sample.get("txt", ""),
+            "tags": sample.get("tags", {}),
+        }
+
+    @staticmethod
+    def _get_bigbio_sample(sample_id, sample) -> dict:
+
+        tags = sample.get("tags", None)
+        if tags:
+            labels = [name + "-" + met_status for name, met_status in tags.items()]
```

Comment:
I think its cleaner if we just include the labels that are ==`met` here ... not an entry for each with `-met` or `-not met` appended. e.g. not this 
```
['ABDOMINAL-not met',
  'ADVANCED-CAD-met',
  'ALCOHOL-ABUSE-not met',
  'ASP-FOR-MI-met',
  'CREATININE-not met',
  'DIETSUPP-2MOS-met',
  'DRUG-ABUSE-not met',
  'ENGLISH-met',
  'HBA1C-not met',
  'KETO-1YR-not met',
  'MAJOR-DIABETES-met',
  'MAKES-DECISIONS-met',
  'MI-6MOS-met']
```
but this 
```
[ 'ADVANCED-CAD',
  'ASP-FOR-MI',
  'DIETSUPP-2MOS',
  'ENGLISH',
  'MAJOR-DIABETES',
  'MAKES-DECISIONS',
  'MI-6MOS']
```

Reply:
I had similar thoughts but wasn't sure about how the bigBio wanted it, better to be more explicit than loosing the information.

* By doing this we're making conscious decision of treating met == presence of `tag` and absence as literally the tag being absent.
* A downside of this approach would be for anyone to find out total unique labels they'd have to iterate all over the dataset and create a set while existing approach simply states presence/absence.
* By having arrays of variable length, it might make it harder to e.g. do one-hot encoding.

I've updated nevertheless as per your suggestion